Branch-SDK/src/io/branch/referral/ServerRequest.java

```diff
@@ -388,41 +388,57 @@ private void updateGAdsParams() {
         BRANCH_API_VERSION version = getBranchRemoteAPIVersion();
         int LATVal = DeviceInfo.getInstance().getSystemObserver().getLATVal();
         String gaid = DeviceInfo.getInstance().getSystemObserver().getGAID();
-        if (!TextUtils.isEmpty(gaid)) {
-            try {
-                if (version == BRANCH_API_VERSION.V2 || version == BRANCH_API_VERSION.V1_CPID ||
-                        version == BRANCH_API_VERSION.V1_LATD) {
-                    JSONObject userDataObj = params_.optJSONObject(Defines.Jsonkey.UserData.getKey());
-                    if (userDataObj != null) {
+        String keyToBeUsedInAdIdsObject = null;
```

Comment:
Let's put this back, and open up a new ticket to refactor this block of code.

Reply:
I couldn't reasonably make the required changes to this method without drastically changing it, even if I kept the existing duplicated code, so I made a gist that explains step by step by how the method morphed into this new version and what REAL changes were made in there vs. just the cleanup changes.
https://gist.github.com/bklastaitis-branch/57c73799d4c1f4144e8bae48122da11a

Reply:
That's awesome!